AppCenterDistribute/AppCenterDistribute/MSDistribute.m

```diff
@@ -228,17 +234,36 @@ - (void)requestInstallInformationWith:(NSString *)releaseHash {
         MSLogDebug([MSDistribute logTag],
                    @"Re-attempting in-app updates setup and cleaning up failure info from storage.");
         [MS_USER_DEFAULTS removeObjectForKey:kMSUpdateSetupFailedPackageHashKey];
+        [MS_USER_DEFAULTS removeObjectForKey:kMSTesterAppUpdateSetupFailedKey];
       }
     }
+    
+    // Create the request ID string and persist it.
+    NSString *requestId = MS_UUID_STRING;
+    [MS_USER_DEFAULTS setObject:requestId forKey:kMSUpdateTokenRequestIdKey];
 
-    NSURL *url;
     MSLogInfo([MSDistribute logTag], @"Request information of initial installation.");
-
-    // Most failures here require an app update. Thus, it will be retried only on next App instance.
-    url = [self buildTokenRequestURLWithAppSecret:self.appSecret releaseHash:releaseHash];
-    if (url) {
-      [self openUrlInAuthenticationSessionOrSafari:url];
-    }
+    
+    // Don't run on the UI thread, or else the app may be slow to startup
+    NSURL *testerAppUrl = [self buildTokenRequestURLWithAppSecret:self.appSecret releaseHash:releaseHash isTesterApp:true];
+    NSURL *installUrl = [self buildTokenRequestURLWithAppSecret:self.appSecret releaseHash:releaseHash isTesterApp:false];
+    dispatch_async(dispatch_get_main_queue(), ^{
+      BOOL shouldUseTesterAppForUpdateSetup = [MS_USER_DEFAULTS objectForKey:kMSTesterAppUpdateSetupFailedKey] == NULL;
+      BOOL testerAppOpened = NO;
+      if (shouldUseTesterAppForUpdateSetup) {
+        MSLogInfo([MSDistribute logTag], @"Attempting to use tester app for update setup.");
+        
+        // Attempt to open the native iOS tester app to enable in-app updates
+        if (testerAppUrl) {
+          testerAppOpened = [self openUrlUsingSharedApp:testerAppUrl];
```

Comment:
Let's log `testerAppOpened` value as Info log in the `if` statement.
We should be able to know what happened in Distribute process from logs, this will say that it found tester app, tried open the app, and succeeded or failed to do it.

Reply:
Very good point!

Reply:
I just saw that you addressed this, Yess!